tmp_file.c

```diff
@@ -0,0 +1,445 @@
+/*
+    tmp_file.c - write to and read from a temporary binary file
+    for fast storage plus added compression.
+    
```

Comment:
This file needs a specific copyright statement and to indicate who the copyright holder is in addition to the author.